src/crypto-bank.Domain/Models/User.cs

```diff
@@ -1,6 +1,35 @@
+using crypto_bank.Domain.Authorization;
+
 namespace crypto_bank.Domain.Models;
 
-public record User(string Email, string Password) //todo: password as plain text is not secure
+public record User
 {
+    public User(int id)
+    {
+        Id = id;
+    }
+
+    public User(
+        string Email,
+        string PasswordHash,
+        string Salt,
+        DateOnly? BirthDate,
+        DateTimeOffset RegisteredAt,
+        Role[] Roles)
+    {
+        this.Email = Email;
+        this.PasswordHash = PasswordHash;
+        this.Salt = Salt;
+        this.BirthDate = BirthDate;
+        this.RegisteredAt = RegisteredAt;
+        this.Roles = Roles;
+    }
+
     public int Id { get; init; }
+    public string Email { get; init; } = null!; // todo: question: is it ok to suppress nullable warning here?
```

Comment:
Yes, EF Core works well with the single constructor. But it doesn't work well when I need to update only one property in an object, because I don't want to use public setters in properties.